main.c

```diff
@@ -0,0 +1,721 @@
+                
+                insertStudent(school, newStudent);
+                break;
+            }
+            case Delete:{
+                // Prompt the user for first name and last name of the student to delete
+                char firstName[50];
+                char lastName[50];
+                
+                fflush(stdin);
+                inputFirstName(firstName);
+                fflush(stdin);
+                inputLastName(lastName);
+                
+                deleteStudentByName(firstName, lastName, school);
+                
+                break;
+            }
+            case Edit:
+                editGrage(school);
+                break;
+            case Search:{
+                // Prompt the user for first name and last name of the student to delete
+                char firstName[50];
+                char lastName[50];
+
+                fflush(stdin);
+                inputFirstName(firstName);
+                fflush(stdin);
+                inputLastName(lastName);
+                
+                dataOfStudent(firstName, lastName, school);
+                break;
+            }
+            case Showall:
+                printAllStudents(school);
+                break;
+            case Top10:{
+                printf("\n\tPlease Enter a Course (0-9): ");
+                int course ;
+                if (scanf("%d", &course) == 1) {
+                    struct TopTenStudents topTenStudent ;
+                    tenBestStudentsOfLayers(school, course, &topTenStudent);
+                    break;
+                }
+                }
+            case UnderperformedStudents:
+                underperformedSstudents(school);
+                break;
+            case Average:{
+                int course;
+                int layer = inputLayer();
+                
+                printf("\n\tPlease Enter a Course (1-10): ");
+                if (scanf("%d", &course) == 1) {
+                    if (course < 1 || course > 10) {
+                        printf("\n\t A wrong course (1-10): ");
+                        break;
+                    }
+                }
+                
+                long average = averageGradesOfCourse(layer, course, school);
+                printf("An average of layer %d of course %d is %ld: ", layer, course, average);
+                break;
+            }
+            case Export:
+                writeAllStudentsToFile(school, "/Users/alex/Documents/ios_projects/check_point_project/check_point_project/students_data.txt");
+                break;
+            case Exit:
+                break;
+            default:
+                printf("\nThere is no item with symbol \"%c\".Please enter a number between 1-10!\nPress any key to continue...",
+                       input);
+                getc(stdin);
+                getc(stdin);
+                break;
+        }
+    } while (input != Exit);
+}
+
+
+int main() {
```

Comment:
Main should always be as clean as possible.
Use functions.

Reply:
Fixed it.